### PR TITLE
Support spaces in run cmd args

### LINKED ISSUE #1859: [BUG]: Support library installs with a path that contains spaces
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem statement

Support library install with a path that contains spaces

### Proposed Solution

In #1853 quotes around the install arguments are introduced, however, the failing integration test in [`test_libraries`](https://github.com/databrickslabs/ucx/blob/main/tests/integration/source_code/test_libraries.py) shows that does not work. Investigate why.

An alternative is to use the `InstallCommand` from `pip._internal.commands` (see [`cells.py`](https://github.com/databrickslabs/ucx/blob/main/src/databricks/labs/ucx/source_code/notebooks/cells.py))

### Additional Context

See [failing test]() introduced by #1853.

### PR DESCRIPTION
**Pull request by ericvergnaud**
## Changes
Current implementation only accepts a full command line, which it splits on spaces. This breaks when the command line contains arguments with spaces, such as `'some stuff'` (note the enclosing quotes)
The new implementation adds support for argument lists, which are passes 'as is' to `Popen`

### Linked issues
Resolves #1859 

### Functionality
None

### Tests
- [x] enabled integration tests

### CODE REVIEW

**Review by JCZuurmond — APPROVED** [review]
lgtm

File: src/databricks/labs/ucx/source_code/python_libraries.py [hunk 1]
```diff
@@ -71,13 +72,14 @@ def _resolve_libraries(path_lookup: PathLookup, *libraries: str) -> list[str]:
         return libs
 
     def _install_pip(self, *libraries: str) -> list[DependencyProblem]:
-        install_command = (
-            f"pip --disable-pip-version-check install {shlex.join(libraries)} -t {self._temporary_virtual_environment}"
-        )
-        return_code, stdout, stderr = self._runner(install_command)
+        args = ["pip", "--disable-pip-version-check", "install"]
+        args.extend(libraries)
+        args.extend(["-t", str(self._temporary_virtual_environment)])
```
Comment (asnare) [inline src/databricks/labs/ucx/source_code/python_libraries.py:77]:
```suggestion
        args = ["pip", "--disable-pip-version-check", "install", *libraries, "-t", str(self._temporary_virtual_environment)]
```

File: src/databricks/labs/ucx/framework/utils.py [hunk 2]
```diff
@@ -29,8 +29,9 @@ def escape_sql_identifier(path: str, optional: bool | None = True) -> str:
     return ".".join(escaped)
 
 
-def run_command(command: str) -> tuple[int, str, str]:
-    logger.info(f"Invoking command: {command}")
-    with subprocess.Popen(command.split(), stdout=subprocess.PIPE, stderr=subprocess.PIPE) as process:
+def run_command(command: str | list[str]) -> tuple[int, str, str]:
+    args = command.split() if isinstance(command, str) else command
+    logger.info(f"Invoking command: {' '.join(args)}")
```
Comment (asnare) [inline src/databricks/labs/ucx/framework/utils.py:34]: I think in terms of understanding _exactly_ what is being executed I'd prefer:
```suggestion
    logger.info(f"Invoking command: {args!r}")
```

**Review by nfx — APPROVED** [review]
lgtm

**Pull request merged**